include/d/d_cs_game.h

```diff
@@ -0,0 +1,353 @@
+            void execute();
+
+        private:
+            STATE_FUNC_DECLARE(lytCrawShotCsr_c, Normal);
+            STATE_FUNC_DECLARE(lytCrawShotCsr_c, ToLock);
+            STATE_FUNC_DECLARE(lytCrawShotCsr_c, Lock);
+            STATE_FUNC_DECLARE(lytCrawShotCsr_c, ToNormal);
+
+            UI_STATE_MGR_DECLARE(lytCrawShotCsr_c);
+            d2d::AnmGroups mAnm;
+            d2d::LytBase_c *mpLyt;
+        };
+
+        /// @brief Gust Bellows cursor.
+        class lytVacuumCsr_c {
+        public:
+            lytVacuumCsr_c(dCsGameAnmGroups_c &g, d2d::LytBase_c *lyt)
+                : mAnm(g.tmp.mAnmGroups, 0x17), mpLyt(lyt), mStateMgr(*this, sStateID::null) {}
+            virtual ~lytVacuumCsr_c() {}
+
+            void init();
+
+            void enter();
+            void execute();
+
+        private:
+            STATE_FUNC_DECLARE(lytVacuumCsr_c, Normal);
+            STATE_FUNC_DECLARE(lytVacuumCsr_c, ToLock);
+            STATE_FUNC_DECLARE(lytVacuumCsr_c, Lock);
+            STATE_FUNC_DECLARE(lytVacuumCsr_c, ToNormal);
+
+            UI_STATE_MGR_DECLARE(lytVacuumCsr_c);
+            d2d::AnmGroups mAnm;
+            d2d::LytBase_c *mpLyt;
+            /* 0x5C */ u8 field_0x5C;
+        };
+
+    public:
+        lytItemCursor_c()
+            : mStateMgr(*this, sStateID::null), mAnm(mAnmGroups.tmp.mAnmGroups, 0x17), mBow(mAnmGroups, &mLyt),
+              mDowsing(mAnmGroups, &mLyt), mPachinko(mAnmGroups, &mLyt), mCrawShot(mAnmGroups, &mLyt),
+              mVacuum(mAnmGroups, &mLyt) {}
+        virtual ~lytItemCursor_c() {}
+        virtual void lytItemCursor0x0C();
+        virtual bool init();
+        virtual bool remove();
+
+        bool preInit();
+        bool doInit();
+
+        void setResAcc(m2d::ResAccIf_c *resAcc) {
+            mpResAcc = resAcc;
+        }
+
+        void setField0x9A0(u8 val) {
+            field_0x9A0 = val;
+        }
+
+        void setNextCursorType(CursorType_e cs) {
+            field_0x9A8 = 1;
+            mNextCursorType = cs;
+        }
+
+        void changeState(const sFStateID_c<lytItemCursor_c> &newState);
+
+    private:
+        STATE_FUNC_DECLARE(lytItemCursor_c, Invisible);
+        STATE_FUNC_DECLARE(lytItemCursor_c, Bow);
+        STATE_FUNC_DECLARE(lytItemCursor_c, Dowsing);
+        STATE_FUNC_DECLARE(lytItemCursor_c, Vacuum);
+        STATE_FUNC_DECLARE(lytItemCursor_c, Pachinko);
+        STATE_FUNC_DECLARE(lytItemCursor_c, HookShot);
+        STATE_FUNC_DECLARE(lytItemCursor_c, PlayerCam);
+
+
+        /* 0x004 */ f32 field_0x004;
+        /* 0x008 */ f32 field_0x008;
+        /* 0x00C */ UI_STATE_MGR_DECLARE(lytItemCursor_c);
+
+        EffectsStruct mEffects;
+        /* 0x070 */ dCsGame_c *mpCsGame;
+        EffectRelated mEffectRelated;
+        /* 0x0C8 */ m2d::ResAccIf_c *mpResAcc;
+        d2d::LytBase_c mLyt;
+        dCsGameAnmGroups_c mAnmGroups;
+        d2d::AnmGroups mAnm;
+        /* 0x??? */ u8 padding[0x44];
+        /* 0x72C */ lytBowCsr_c mBow;
```

Comment:
Im starting to notice not commenting offsets. 
Should we keep commenting offsets or now due to clangd, should we not?

Reply:
That was just me being lazy. I like hex offsets (outside of small RELs where they're probably not super necessary), and clangd only gives them in decimal so the hex offsets would still be nice here. I couldn't be bothered to math out the offsets in the inlined ctor, and I wasn't quite certain about which members belonged to classes with more inlined ctors and which members are standalone.

Reply:
Added offsets, turns out I had one mistake around the EffectsStruct, which holds a dBase_c* or fBase_c* or void*, not a dAcBase_c*.